include/oneapi/tbb/blocked_nd_range.h

```diff
@@ -147,6 +147,36 @@ class blocked_nd_range : public blocked_nd_range_impl<Value, N> {
     using base::base;
 };
 
+#if __TBB_CPP17_DEDUCTION_GUIDES_PRESENT
+template <typename Value, unsigned int... Ns,
+          typename = std::enable_if_t<(... && (Ns == 2 || Ns == 3))>>
+blocked_nd_range(const Value (&... dim)[Ns])
+-> blocked_nd_range<Value, sizeof...(Ns)>;
+
+template <typename Value, typename... Values,
+          typename = std::enable_if_t<(... && std::is_same_v<Value, Values>)>>
+blocked_nd_range(blocked_range<Value>, blocked_range<Values>...)
+-> blocked_nd_range<Value, 1 + sizeof...(Values)>;
+
+template <typename Value, unsigned int N,
+          typename = std::enable_if_t<(N != 2 && N != 3)>>
+blocked_nd_range(const Value (&)[N])
+-> blocked_nd_range<Value, N>;
+
+template <typename Value, unsigned int N>
+blocked_nd_range(const Value (&)[N], typename blocked_nd_range<Value, N>::size_type)
+-> blocked_nd_range<Value, N>;
+
+template <typename Value, unsigned int N>
+blocked_nd_range(blocked_nd_range<Value, N>, oneapi::tbb::split)
+-> blocked_nd_range<Value, N>;
+
+template <typename Value, unsigned int N>
+blocked_nd_range(blocked_nd_range<Value, N>, oneapi::tbb::proportional_split)
+-> blocked_nd_range<Value, N>;
+
```

Comment:
Splitting constructors are not really for users but for algorithms. Do we need deduction guides for these?